cpp/src/arrow/array/builder_dict.h

```diff
@@ -316,14 +309,14 @@ template <typename BuilderType>
 class DictionaryBuilderBase<BuilderType, NullType> : public ArrayBuilder {
  public:
   DictionaryBuilderBase(const std::shared_ptr<DataType>& value_type,
-                        MemoryPool* pool = default_memory_pool())
```

Comment:
I thought we were talking about getting rid of this, but I opened https://issues.apache.org/jira/browse/ARROW-7256

Reply:
Reverted in this changeset.

Reply:
Yanked it from the PR.